File: music-infrastructure/src/main/java/com/music/sale/persistence/store/entity/StoreEntity.java
package com.music.sale.persistence.store.entity;

import com.music.sale.persistence.common.BaseEntity;
import jakarta.persistence.*;

@Entity
@Table(name = "store")
public class StoreEntity extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, length = 1000)
    private String description;

    @Column(nullable = false)
    private String zipcode;

    @Column(nullable = false)
    private String baseAddress;

    @Column(nullable = true)
    private String detailAddress;

    @Column(nullable = true)
    private Double latitude;

    @Column(nullable = true)
    private Double longitude;

    @Column(nullable = false)
    private String contactNumber;

    @Column(nullable = false)
    private String businessNumber;

    @Column(nullable = true)
    private String imageUrl;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private StoreStatus status;

    @Column(nullable = false)
    private Long sellerId;

    // 기본 생성자
    public StoreEntity() {}

    // 모든 필드 생성자
    public StoreEntity(Long id, String name, String description, String zipcode,
                       String baseAddress, String detailAddress, Double latitude,
                       Double longitude, String contactNumber, String businessNumber,
                       String imageUrl, StoreStatus status, Long sellerId) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.zipcode = zipcode;
        this.baseAddress = baseAddress;
        this.detailAddress = detailAddress;
        this.latitude = latitude;
        this.longitude = longitude;
        this.contactNumber = contactNumber;
        this.businessNumber = businessNumber;
        this.imageUrl = imageUrl;
        this.status = status;
        this.sellerId = sellerId;
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getZipcode() { return zipcode; }
    public void setZipcode(String zipcode) { this.zipcode = zipcode; }

    public String getBaseAddress() { return baseAddress; }
    public void setBaseAddress(String baseAddress) { this.baseAddress = baseAddress; }

    public String getDetailAddress() { return detailAddress; }
    public void setDetailAddress(String detailAddress) { this.detailAddress = detailAddress; }

    public Double getLatitude() { return latitude; }
    public void setLatitude(Double latitude) { this.latitude = latitude; }

    public Double getLongitude() { return longitude; }
    public void setLongitude(Double longitude) { this.longitude = longitude; }

    public String getContactNumber() { return contactNumber; }
    public void setContactNumber(String contactNumber) { this.contactNumber = contactNumber; }

    public String getBusinessNumber() { return businessNumber; }
    public void setBusinessNumber(String businessNumber) { this.businessNumber = businessNumber; }

    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }

    public StoreStatus getStatus() { return status; }
    public void setStatus(StoreStatus status) { this.status = status; }

    public Long getSellerId() { return sellerId; }
    public void setSellerId(Long sellerId) { this.sellerId = sellerId; }

    public enum StoreStatus {
        PENDING,
        ACTIVE,
        SUSPENDED,
        CLOSED
    }
}

File: music-infrastructure/src/main/kotlin/com/music/sale/persistence/store/entity/StoreEntity.kt
// Copyright (C) 2024 Your Name or Company
package com.music.sale.persistence.store.entity

import com.music.sale.persistence.common.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "store")
class StoreEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) val id: Long = 0L,
    @Column(nullable = false) val name: String,
    @Column(nullable = false, length = 1000) val description: String,
    @Column(nullable = false) val zipcode: String,
    @Column(nullable = false) val baseAddress: String,
    @Column(nullable = true) val detailAddress: String? = null,
    @Column(nullable = true) val latitude: Double? = null,
    @Column(nullable = true) val longitude: Double? = null,
    @Column(nullable = false) val contactNumber: String,
    @Column(nullable = false) val businessNumber: String,
    @Column(nullable = true) val imageUrl: String? = null,
    @Column(nullable = false) @Enumerated(EnumType.STRING) val status: StoreStatus,
    @Column(nullable = false) val sellerId: Long,
) : BaseEntity() {
    enum class StoreStatus {
        PENDING,
        ACTIVE,
        SUSPENDED,
        CLOSED,
    }
}
